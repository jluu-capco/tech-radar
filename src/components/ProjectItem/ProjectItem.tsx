import React, { FC } from "react";
import ProjectTechnologyLink from "../ClinetProjectLink/ProjectTechnologyLink";
import images from "../../images/ClientProjects";
import StyledWrapper from "./styles";

type ProjectItemProps = {
  data: Array<{
    project: string;
    clientName?: string;
    clientImage: string;
    description: string;
    technologies: Array<string>;
  }>;
  githubLink?: string;
};

const ProjectItem: FC<ProjectItemProps> = ({ data, githubLink }) => (
  <>
    {data.map(
      (
        { project, clientName, clientImage, description, technologies },
        index
      ) => (
        <StyledWrapper
          key={`project-${index}`}
          data-test-id={`project-${index}`}
        >
          <div>
            <div className="project-image-wrapper">
              <img
                className="project-image"
                src={(images as any)[clientImage]}
                alt="computer screen"
              />
            </div>
            <div className="row-wrapper">
              <div className="heading">{project}</div>
              {githubLink && (
                <a href="#" className="link">
                  Github Link
                </a>
              )}
              {clientName ? (
                <div className="project-tag">{clientName}</div>
              ) : (
                <div className="project-tag">Internal</div>
              )}
            </div>
            <div className="text">{description}</div>
            <div className="heading">Key Technologies</div>
            <div className="technologies">
              {technologies.map((tech: string, idx: number) => (
                <span key={`tech-${idx}`}>
                  <ProjectTechnologyLink techName={tech} />
                </span>
              ))}
            </div>
          </div>
        </StyledWrapper>
      )
    )}
  </>
);

export default ProjectItem;
