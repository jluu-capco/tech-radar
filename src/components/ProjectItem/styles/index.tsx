import styled from "styled-components/macro";

const StyledWrapper = styled.div`
    margin-bottom: 20px;
    font-family: poppins, sans-serif;
  .projectTech-link {
    display: inline-block;
    padding 10px;
    color: #ffffff;
    text-align: center;
  }
  svg.icon {
    height: 45px;
    width: 45px;
  }
  .div {
    margin-bottom: 10px;
  }
  .project-image-wrapper {
    padding: 5% 10%;
    background-color: white;
    margin-bottom: 20px;
    border-radius: 4px;
  }
  .project-image {
    width: 100%;
  }
  .heading {
    font-weight: 700;
    font-size: 18px;
    line-height: 27px;
    margin-bottom: 12px;
  }
  .link {
    grid-column: 2 / span 1;
    grid-row: 1 / span 2;
    justify-self: end;
    align-self: center;
    color: white;
    font-size: 16px;
    line-height: 24px;
    border-bottom: 1px solid #e6236d;
    padding-bottom: 4px;
  }
  .link:hover {
    cursor: pointer;
  }
  .project-tag {
    grid-column: 1 / span 1;
    background-color: rgba(230, 35, 109, 0.5);
    border-radius: 40px;
    min-width: min-content;
    max-width: max-content;
    padding: 4px 16px;
    text-align: center;
    align-self: center;
    font-weight: 500;
    font-size: 14px
    line-height: 21px;
  }
  .text {
    margin-bottom: 20px;
    font-size: 14px
    line-height: 21px;
  }
  .technologies {
    margin-bottom: 100px;
  }
  .row-wrapper {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: repeat(2, 1fr);
    margin-bottom: 20px;
  }
}
`;

export default StyledWrapper;
