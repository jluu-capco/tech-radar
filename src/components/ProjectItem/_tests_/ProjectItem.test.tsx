import * as React from "react";
import { render, screen } from "@testing-library/react";
import { RadarContextProvider } from "../../RadarContextProvider/RadarContextProvider";
import ProjectItem from "../ProjectItem";
import { BrowserRouter as Router } from "react-router-dom";

window.scroll = jest.fn();
const mockValue = [
  {
    project: "GLOBAL DevOpS Acceleration",
    clientName: "Global Tier 1 Private Bank",
    clientImage: "CL1",
    description:
      "Capco delivered a DevOps acceleration program focused on improving delivery quality and speed.  This included integrating existing tools to improve developer experience, simplifying consumption of existing internal Kubernetes services, cloud migration, uplifting the platforms observability and monitoring maturity, re-engineering for containerization, building a single pane of glass portal, and training global teams e.g., India, China.",
    technologies: ["Go", "Docker", "AWS", "Kubernetes", "Datadog"],
  },
];
describe("ProjectItem", () => {
  it("should render ProjectItem component", () => {
    const { container } = render(
      <Router>
        <RadarContextProvider>
          <ProjectItem data={mockValue} />
        </RadarContextProvider>
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render GitHub ProjectItem component and check the correct text is displayed", () => {
    render(
      <Router>
        <RadarContextProvider>
          <ProjectItem data={mockValue} />
        </RadarContextProvider>
      </Router>
    );

    expect(screen.getByTestId("project-0").focus).toBeTruthy();
  });
});
