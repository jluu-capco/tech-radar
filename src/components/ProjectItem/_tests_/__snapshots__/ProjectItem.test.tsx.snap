// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem should render ProjectItem component 1`] = `
<div
  class="styles__StyledWrapper-sc-1pgv985-0 rxkTS"
  data-test-id="project-0"
>
  <div>
    <div
      class="project-image-wrapper"
    >
      <img
        alt="computer screen"
        class="project-image"
        src="CL1.png"
      />
    </div>
    <div
      class="row-wrapper"
    >
      <div
        class="heading"
      >
        GLOBAL DevOpS Acceleration
      </div>
      <div
        class="project-tag"
      >
        Global Tier 1 Private Bank
      </div>
    </div>
    <div
      class="text"
    >
      Capco delivered a DevOps acceleration program focused on improving delivery quality and speed.  This included integrating existing tools to improve developer experience, simplifying consumption of existing internal Kubernetes services, cloud migration, uplifting the platforms observability and monitoring maturity, re-engineering for containerization, building a single pane of glass portal, and training global teams e.g., India, China.
    </div>
    <div
      class="heading"
    >
      Key Technologies
    </div>
    <div
      class="technologies"
    >
      <span>
        <a
          class="projectTech-link"
          href="/technology/backend/go"
        >
          <svg
            class="icon"
            viewBox="0 0 80 80"
          >
            <circle
              cx="40"
              cy="40"
              fill="white"
              r="40"
            />
            <image
              height="50"
              href="https://img.icons8.com/color/344/golang.png"
              width="50"
              x="15"
              y="15"
            />
          </svg>
          <div>
            Go
          </div>
        </a>
      </span>
      <span>
        <a
          class="projectTech-link"
          href="/technology//docker"
        >
          <svg
            class="icon"
            viewBox="0 0 80 80"
          >
            <circle
              cx="40"
              cy="40"
              fill="white"
              r="40"
            />
            <image
              height="50"
              href=""
              width="50"
              x="15"
              y="15"
            />
          </svg>
          <div>
            Docker
          </div>
        </a>
      </span>
      <span>
        <a
          class="projectTech-link"
          href="/technology/cloud-&-platforms/aws"
        >
          <svg
            class="icon"
            viewBox="0 0 80 80"
          >
            <circle
              cx="40"
              cy="40"
              fill="white"
              r="40"
            />
            <image
              height="50"
              href="https://img.icons8.com/color/452/amazon-web-services.png"
              width="50"
              x="15"
              y="15"
            />
          </svg>
          <div>
            AWS
          </div>
        </a>
      </span>
      <span>
        <a
          class="projectTech-link"
          href="/technology/cloud-&-platforms/kubernetes"
        >
          <svg
            class="icon"
            viewBox="0 0 80 80"
          >
            <circle
              cx="40"
              cy="40"
              fill="white"
              r="40"
            />
            <image
              height="50"
              href="https://img.icons8.com/color/452/kubernetes.png"
              width="50"
              x="15"
              y="15"
            />
          </svg>
          <div>
            Kubernetes
          </div>
        </a>
      </span>
      <span>
        <a
          class="projectTech-link"
          href="/technology//datadog"
        >
          <svg
            class="icon"
            viewBox="0 0 80 80"
          >
            <circle
              cx="40"
              cy="40"
              fill="white"
              r="40"
            />
            <image
              height="50"
              href=""
              width="50"
              x="15"
              y="15"
            />
          </svg>
          <div>
            Datadog
          </div>
        </a>
      </span>
    </div>
  </div>
</div>
`;
